### ISSUE TITLE
magnet hosts are discarded

### ISSUE THREAD

**Issue by hboetes** [aria2/aria2#727]
I noticed that the hosts found when downloading the bittorrent file from the magnet link are discarded and the tracker is consulted again. Wouldn't it be clever to keep those hosts in memory somewhere?

**Comment by tatsuhiro-t** [comment]
Do you mean "host" is included in magnet link?  I don't know about such parameter.

**Comment by hboetes** [comment]
No, the magnet link includes trackers, the trackers send aria2c hosts which have the bittorrent file. They are the same hosts that have the requested data as well, Unless I'm totally mistaken.

In the current set up, the trackers are asked twice for the same hosts.

**Comment by tatsuhiro-t** [comment]
Ah, so you argues that aria2 discards all peers after bittorrent metadata is downloaded, and then contact tracker again to download content described in bittorrent metadata.
Yes, it is how aria2 performs right now because of architectural limitation.
It would be good enhancement.

**Comment by hboetes** [comment]
Exactly! Of course it's not a big issue, just nice to have. I'm glad you
take my request into consideration.
